### ISSUE TITLE
Disable component types that are not used in a collection when run from the Editor

### ISSUE THREAD

**Issue by AGulev** [defold/defold#6668]
**Task (REQUIRED):**
The Bob tool currently counts components in collection and its factories, to be able to save memory in runtime.
The editor doesn't do anything like that.
That would be nice if the editor does the same. It would help with memory profiling and will make build result more consistent.
https://github.com/defold/defold/blob/6b03d6f75afb1411f62d96850cd8d94b1971916e/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/CollectionBuilder.java#L229-L255

**Expected outcome (REQUIRED):**
The editor counts component in collection.

PR for BOB: https://github.com/defold/defold/pull/6062
Task: https://github.com/defold/defold/issues/5843

**Comment by Vbif** [comment]
Is there any plans for this issue? 

As https://github.com/defold/defold/issues/5843 says most games with physics need only one physics world. But in term of collections situation often looks like this: bootstrap -> title -> level select -> game. And you need define at least 4 phy world in game.project to run a game. And if you dont want to see "ERROR:PHYSICS" messages on every popup you need increase number of worlds even more.

**Comment by britzl** [comment]
> Is there any plans for this issue?

Yes, this is still something we plan to implement also in the editor at some point. Although right now we are focusing a lot of effort on performance related improvements for the editor when working with large projects so I can't say for sure when it will happen.

**Comment by AGulev** [comment]
This is the reason why user may have "Physics world buffer full, world could not be created." even when no physics components were used in the project
